src/Marten/Internal/UnitOfWork.cs

```diff
@@ -219,6 +219,17 @@ public void Eject<T>(T document)
             }
         }
 
+        public void EjectAllOfType(Type type)
+        {
+            var operations = operationsFor(type);
+            var matching = operations.OfType<IDocumentStorageOperation>().ToArray();
+
+            foreach (var operation in matching)
```

Comment:
You have to check whether the `IDocumentStorageOperation` involves the `type` here.

Reply:
@jeremydmiller  `var operations = operationsFor(type);`, this line does that check, anything else needed?

Reply:
Ah, sorry, I missed that. Still need a test to prove that it's filtering correctly though.

Reply:
Good point! Hopefully I have time later this week to add more tests!